policies/rudderc/src/backends/metadata.rs

```diff
@@ -4,87 +4,298 @@
+        }
+    }
+}
+
+// For examples of serde serialization in quick-xml:
+// https://github.com/tafia/quick-xml/blob/master/tests/serde-se.rs
+
 #[derive(Debug, PartialEq, Serialize)]
 struct Technique {
     name: String,
-    description: Description,
+    description: Option<Description>,
+    #[serde(rename = "usemethodreporting")]
+    use_method_reporting: UseMethodReporting,
+    agent: Vec<Agent>,
+    sections: Sections,
 }
 
 #[derive(Debug, PartialEq, Serialize)]
 struct Description {
+    #[serde(rename = "$value")]
+    value: String,
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+struct UseMethodReporting {
+    #[serde(rename = "$value")]
+    value: bool,
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+struct Agent {
+    #[serde(rename = "type")]
+    _type: String,
+    bundles: Bundles,
+    files: Files,
+}
+
+impl Agent {
+    /// Types used in metadata.xml
+    fn type_from_target(t: Target) -> &'static str {
+        match t {
+            Target::Unix => "cfengine-community,cfengine-nova",
+            Target::Windows => "dsc",
+            Target::Docs | Target::Metadata => unreachable!(),
+        }
+    }
+
+    fn from(_resources: Vec<ResourceKind>, target: Target) -> Agent {
+        Agent {
+            _type: Self::type_from_target(target).to_string(),
+            // FIXME we need bundle names from methods metadata
+            bundles: Bundles {
+                name: vec![Name {
+                    value: "package_present".to_string(),
+                }],
+            },
+            files: Files { file: vec![] },
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+struct Files {
+    file: Vec<File>,
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+struct File {
+    name: String,
+    included: Included,
+    #[serde(rename = "outpath")]
+    out_path: Option<OutPath>,
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+struct Included {
+    #[serde(rename = "$value")]
+    value: bool,
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+struct OutPath {
+    #[serde(rename = "$value")]
     value: String,
 }
 
+#[derive(Debug, PartialEq, Serialize)]
+struct Bundles {
+    name: Vec<Name>,
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+struct Name {
+    #[serde(rename = "$value")]
+    value: String,
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+struct Sections {
+    section: Vec<Section>,
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+struct Section {
+    name: String,
+    id: Id,
+    component: bool,
+    multivalued: bool,
+    value: Vec<ReportKey>,
+    section: Vec<Section>,
+}
+
+impl Section {
+    fn from(resources: Vec<ResourceKind>) -> Vec<Self> {
+        // recursive function to handle blocks. Block recursion levels shouldn't be a
+        // stack overflow threat.
+        resources
+            .into_iter()
+            .map(|r| match r {
+                ResourceKind::Block(b) => Section {
+                    name: b.name,
+                    id: b.id,
```

Comment:
Why is this the only id that is not cloned?

Reply:
Because it is moved (as we do not reuse it after)